### PR TITLE
env:Improve GNU coreutils Compatibility & Fix env-signal-handler.sh Test

### PR DESCRIPTION
**Pull request by mattsu2020**
### Key Improvements

Increased GNU coreutils compatibility by aligning behavior with upstream implementations and addressing differences in edge-case handling.
Fixed the env-signal-handler.sh test, ensuring correct signal propagation and consistent behavior across platforms.
Added minor adjustments to internal logic to better emulate GNU semantics and reduce divergence.

### Why This Matters

Improving compatibility helps maintain predictable behavior for users migrating from GNU coreutils. Successfully passing the env-signal-handler.sh test strengthens signal-handling reliability and contributes to overall test suite stability.

### CODE REVIEW

**Comment by ChrisDryden** [comment]
Any chance theres a clean way to split the compatibility fixes from the test fixes? Its just quite a big CR to review and its hard to match the tests to the code changes. Maybe at least splitting the changes in the other bins?

**Comment by mattsu2020** [comment]
> Any chance theres a clean way to split the compatibility fixes from the test fixes? Its just quite a big CR to review and its hard to match the tests to the code changes. Maybe at least splitting the changes in the other bins?

separate this PR
seq only
https://github.com/uutils/coreutils/pull/9471

File: src/uu/seq/src/seq.rs [hunk 1]
```diff
@@ -212,6 +212,7 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
     match result {
         Ok(()) => Ok(()),
         Err(err) if err.kind() == std::io::ErrorKind::BrokenPipe => {
+
```
Comment (sylvestre) [inline src/uu/seq/src/seq.rs]: please remove this change

Comment (mattsu2020) [reply src/uu/seq/src/seq.rs]: sorry fix

**Comment by collinfunk** [comment]
I'm confused. Why is the support for `__ALL__` needed?

**Comment by oech3** [comment]
@mattsu2020 https://github.com/uutils/coreutils/pull/9567#issuecomment-3693796828
